core/arch/riscv/include/kernel/thread.h

```diff
@@ -0,0 +1,180 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
```

Comment:
This files duplicates quite a few things from `core/arch/arm/include/kernel/thread.h`. Isn't there a way to use a common file for non arch-specific things?